Automattic/jetpack
# Quality of life updates for the Recipe block

Partial Fixes for #24632 

#### Changes proposed in this Pull Request:
* Adds block icons for the Recipe block, and sub-blocks.
* Adds functionality to enable users to hit "return" to add a new ingredient to the recipe.
* Adds functionality to enable users to delete the recipe item block after they've deleted the content of the recipe item.

#### Does this pull request change what data or activity we track or use?

My PR does not change what data or activity is tracked or used.

#### Other information:


#### Testing instructions:

* Add the Recipe block to a new or existing post, or page.
* Using the block navigator, you should see unique icons for each sub-block pertaining to the recipe.
* When adding ingredients to the list, you may hit return within an existing ingredient, and a new ingredient item will be appended to the end of the list.
* When editing an existing ingredient in the list, deleting the content, and then hitting delete once more time will remove the single ingredient item altogether.
